### PR TITLE
Adds admin freezing

### PR DESCRIPTION
**Pull request by FlattestGuitar**
Right click menu and stuff

All admins get it

Non-admins don't

Beautiful sprites of being ~~b&~~ reprimanded from Paradise Station.

### CODE REVIEW

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -472,6 +472,30 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+	if(!holder)
+		to_chat(src, "Only administrators may use this command.")
+		return
+	if(!istype(M))
+		return
+
+	if(!M.adminfrozen)
+		var/sleepies = M.AmountSleeping()
+		M.adminfrozen = sleepies ? sleepies : 1
+		M.SetSleeping(20000)//2k seconds to get your admin shit together
```
Comment (AndrewMontagne) [inline code/modules/admin/verbs/randomverbs.dm]: `INFINITY` instead of `20000`?

Comment (FlattestGuitar) [reply code/modules/admin/verbs/randomverbs.dm]: Not really necessary. 2k will be more than enough and things might get unexpectedly funky.

Comment (SamHPurp) [reply code/modules/admin/verbs/randomverbs.dm]: 2k is 33 mins.

File: code/modules/admin/verbs/randomverbs.dm [hunk 2]
```diff
@@ -472,6 +472,30 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+		M.adminfreezeoverlay = new()
+		M.add_overlay(M.adminfreezeoverlay)
+		log_admin("[key_name(usr)] froze [key_name(M)]!")
+		to_chat(M, "<span class='userdanger'>You have been frozen by Administrator [usr.key]!</span>")
```
Comment (SamHPurp) [inline code/modules/admin/verbs/randomverbs.dm]: Maybe increase the size yo

File: code/game/objects/effects/overlays.dm [hunk 3]
```diff
@@ -43,3 +43,8 @@
 	name = "sparkles"
 	icon = 'icons/effects/effects.dmi'
 	icon_state = "shieldsparkles"
+
+/obj/effect/overlay/bad_boy
```
Comment (SamHPurp) [inline code/game/objects/effects/overlays.dm]: better name plz

Comment (Ziiro) [reply code/game/objects/effects/overlays.dm]: ~naughty player~
🤔

File: code/modules/admin/verbs/randomverbs.dm (hunk 2 above)
Comment (FlattestGuitar) [reply code/modules/admin/verbs/randomverbs.dm]: Doesn't get bigger than this without becoming unreadable. `userdanger` is what spams so hard when you're attacked. As high as a quarter of your chat window. Even worse on smaller screens.

Comment (SamHPurp) [reply code/modules/admin/verbs/randomverbs.dm]: oh right, cool

File: code/modules/admin/verbs/randomverbs.dm [hunk 4]
```diff
@@ -484,7 +484,7 @@ Traitors and the like can also be revived with the previous role mostly intact.
 	if(!M.adminfrozen)
 		var/sleepies = M.AmountSleeping()
 		M.adminfrozen = sleepies ? sleepies : 1
-		M.SetSleeping(20000)//2k seconds to get your admin shit together
+		M.SetSleeping(200000)//20k seconds to get your admin shit together
```
Comment (SamHPurp) [inline code/modules/admin/verbs/randomverbs.dm]: Why, I mean, fine, but 2k was fine... 2k is literally a bit more than 30 mins.... 20k is 5 and a half hours of frozen time.

**Comment by AndrewMontagne** [comment]
Resolved the merge conflict for you, @FlattestGuitar.

**Comment by SamHPurp** [comment]
[Add 'Admin Freezing'](https://trello.com/c/A9jJzB8S/98-add-admin-freezing)

File: code/modules/admin/verbs/randomverbs.dm [hunk 5]
```diff
@@ -472,6 +472,30 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+		return
+
+	if(!M.adminfrozen)
+		var/sleepies = M.AmountSleeping()
+		M.adminfrozen = sleepies ? sleepies : 1
+		M.SetSleeping(200000)//20k seconds to get your admin shit together
+		M.adminfreezeoverlay = new()
+		M.add_overlay(M.adminfreezeoverlay)
+		log_admin("[key_name(usr)] froze [key_name(M)]!")
+		to_chat(M, "<span class='userdanger'>You have been frozen by Administrator [usr.key]!</span>")
```
Comment (SamHPurp) [inline code/modules/admin/verbs/randomverbs.dm]: by `an` administrator

File: code/modules/admin/verbs/randomverbs.dm [hunk 6]
```diff
@@ -472,6 +472,30 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+	else
+		M.SetSleeping(M.adminfrozen)//set it to what it was before freezing or just 1/10th of a second if it was nothing
+		M.adminfrozen = 0
+		M.cut_overlay(M.adminfreezeoverlay)
+		log_admin("[key_name(usr)] unfroze [key_name(M)].")
+		to_chat(M, "<span class='userdanger'>You have been unfrozen by Administrator [usr.key]!</span>")
```
Comment (SamHPurp) [inline code/modules/admin/verbs/randomverbs.dm]: by `an` administrator

File: code/modules/admin/verbs/randomverbs.dm (hunk 5 above)
Comment (FlattestGuitar) [reply code/modules/admin/verbs/randomverbs.dm]: It's `You have been frozen by Administrator Purpose2`. Not `an Administrator Purpose2`

Comment (SamHPurp) [reply code/modules/admin/verbs/randomverbs.dm]: o. rite.

**Pull request merged**